src/Action/CmdLine.hs

```diff
@@ -80,7 +82,12 @@ data CmdLine
 
 defaultDatabaseLang :: Language -> IO FilePath
 defaultDatabaseLang lang = do
-    dir <- getAppUserDataDirectory "hoogle"
+    xdgLocation <- getXdgDirectory XdgData "hoogle"
+    legacyLocation <- getAppUserDataDirectory "hoogle"
+    canIgnoreLegacyPath <- not <$> doesPathExist legacyLocation
+    unless canIgnoreLegacyPath $
```

Comment:
Could we avoid double negation? I'd write `doesLegacyPathExist <- doesPathExist legacyLocation` followed by `when doesLegacyPathExist`.